src/main/java/org/myrobotlab/service/Arduino.java

```diff
@@ -471,6 +472,26 @@ public void reattach(DeviceMapping dm) {
       if (servo.isEnabled()) {
         msg.servoAttachPin(dm.getId(), pin);
       }
+    } else if (attachable instanceof UltrasonicSensorControl) {
+      // reattach logic
+      // } else if (attachable instanceof Pir) { Pir is a PinListener
+      // reattach logic - FIXME Pir has no Control interface :(
+    } else if (attachable instanceof PinListener) {
+      PinListener pl = (PinListener) attachable;
+      attachPinListener((PinListener) pl);
+
+      // on reattach get back to its previous state enabled/disabled
+      if (attachable instanceof Pir) {
+        Pir pir = (Pir) attachable;
+        if (pir.isEnabled()) {
+          pir.enable();
+        }
+      }
+
+    } else if (attachable instanceof I2CControl) {
+

```

Comment:
no implementation here.. a log message should be added to say this isn't supported yet.